03-path-application-development/305-app-scaling-custom-metrics/readme.adoc

```diff
@@ -0,0 +1,206 @@
+= Kubernetes App Auto-scaling with Custom metrics
+:toc:
+:icons:
+:linkcss:
+:imagesdir: ../../resources/images
+
+https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/[Horizontal Pod Autoscaling] (HPA) is a Kubernetes feature to dynamically increase/decrease the number of pod replicas based on resource utilization metrics.
+
+The Horizontal Pod Autoscaling feature was introduced in Kubernetes v1.2. It allows users to autoscale off of basic metrics like CPU, but requires a resource called metrics-server to run along side your application. As of Kubernetes v1.6, it is possible to autoscale off of custom metrics. Custom metrics are user defined and are collected from within the cluster. As of Kubernetes v1.10, support for external metrics was introduced so users can autoscale off of any metric from outside the cluster that is collected for you by Datadog.
+
+The custom and external metric providers, as opposed to the metrics server, are resources that have to be implemented and registered by the user.
+
+== Prerequisites
+
+Running Kubernetes v1.10+ in order to be able to register the External Metrics Provider resource against the API Server.
+Having the Aggregation layer enabled, refer to the https://kubernetes.io/docs/tasks/access-kubernetes-api/configure-aggregation-layer/[Kubernetes aggregation layer] configuration documentation to learn how to enable it.
+Using EKS 2.0, this will be automatically enabled for you.
```

Comment:
Can you link this to EKS platform versions documentation?

Using link:[EKS 2.0]https://docs.aws.amazon.com/eks/latest/userguide/platform-versions.html, this will be ...